### PR TITLE
Throw an exception if Netbios name exceeds 15 characters

### PR DESCRIPTION
**Pull request by indualagarsamy**
Fixes https://github.com/Particular/NServiceBus.Transport.Msmq/issues/14

NetBios name has to be 15 characters or less for MSMQ to work. 
See: http://geekswithblogs.net/Plumbersmate/archive/2012/02/03/make-sure-computer-names-are-15-characters-or-less-fro.aspx

### CODE REVIEW

File: src/NServiceBus.Transport.Msmq/CheckMachineNameForComplianceWithDtcLimitation.cs [hunk 1]
```diff
@@ -1,18 +1,18 @@
 namespace NServiceBus.Transport.Msmq
 {
+    using System;
     using System.Runtime.InteropServices;
     using System.Text;
-    using Logging;
 
-    class CheckMachineNameForComplianceWithDtcLimitation
+    static class CheckMachineNameForComplianceWithDtcLimitation
```
Comment (indualagarsamy) [inline src/NServiceBus.Transport.Msmq/CheckMachineNameForComplianceWithDtcLimitation.cs]: I think class name be changed. I thought this was more of a MSMQ limitation than a DTC limitation. See John Breakwell's article. Thoughts @Particular/msmq-transport-maintainers ?

**Comment by indualagarsamy** [comment]
@Particular/msmq-transport-maintainers - please review. cc / @dvdstelt

File: src/NServiceBus.Transport.Msmq/CheckMachineNameForCompliance.cs [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
-    class CheckMachineNameForComplianceWithDtcLimitation
+    static class CheckMachineNameForCompliance
     {
         [DllImport("kernel32.dll", SetLastError = true, CharSet = CharSet.Auto)]
         static extern bool GetComputerNameEx(COMPUTER_NAME_FORMAT nameType, [Out] StringBuilder lpBuffer, ref uint lpnSize);
 
         /// <summary>
         /// Method invoked to run custom code.
         /// </summary>
-        public void Check()
```
Comment (SimonCropp) [inline src/NServiceBus.Transport.Msmq/CheckMachineNameForCompliance.cs]: pretty sure the above doco comment adds no value ?

File: src/NServiceBus.Transport.Msmq/CheckMachineNameForCompliance.cs [hunk 3]
```diff
@@ -25,12 +25,9 @@ public void Check()
             {
                 return;
             }
-
-            Logger.WarnFormat("NetBIOS name [{0}] is longer than 15 characters. Shorten it for DTC to work.", netbiosName);
+            throw new InvalidOperationException($"The NetBIOS name {netbiosName} is longer than 15 characters. Shorten the machine name to 15 characters or less for MSMQ to deliver messages.");
```
Comment (SimonCropp) [inline src/NServiceBus.Transport.Msmq/CheckMachineNameForCompliance.cs]: no value in a `throw new InvalidOperationException`. a `throw new Exception` will suffice

File: src/NServiceBus.Transport.Msmq/CheckMachineNameForCompliance.cs (hunk 2 above)
Comment (indualagarsamy) [reply src/NServiceBus.Transport.Msmq/CheckMachineNameForCompliance.cs]: @SimonCropp - yeah, I agree. Although I didn't modify it, it makes sense to change it as part of this PR.

**Comment by indualagarsamy** [comment]
@SimonCropp - I've addressed your comments.

**Pull request merged**